### ISSUE TITLE
Invalid "Remove Unnecessary Cast" suggestion when casting any numeric type to object

### ISSUE THREAD

**Issue by ygoe** [dotnet/roslyn#26640]
**Version Used**: VS 2017 15.6.7

**Steps to Reproduce**:

```
object Method1(bool b) => b ? (byte)1 : (byte)0;

byte Method2() => (byte)Method1(true);
```

**Expected Behavior**:

No warnings.

**Actual Behavior**:

The two cast operators in Method1 are flagged as unnecessary. But they actually do have an effect. The code works as is. Without these cast operators, the cast in Method2 would fail because the value is int not byte.

**Comment by CyrusNajmabadi** [comment]
> The two cast operators in Method1 are flagged as unnecessary. But they actually do have an effect. The code works as is. Without these cast operators, the cast in Method2 would fail because the value is int not byte.

Both are flagged, because *individually* neither is necessary.  However, if you fix, it will only remove one.  If you fix-all, it shoudl also only remove one.  If you see otherwise, let us know.

**Comment by sharwell** [comment]
@CyrusNajmabadi *both* casts are required in this code. If you remove _either_ cast, it will box `Int32`

**Comment by CyrusNajmabadi** [comment]
> @CyrusNajmabadi both casts are required in this code. If you remove either cast, it will box Int32

Good call.  I was thinking about the just the 'int' case.  But you're correct.  In the narrowing conversion case, we'd need the cast on both sides.
